modules/openapi-generator/src/test/java/org/openapitools/codegen/TestUtils.java

```diff
@@ -28,6 +28,30 @@
 
 public class TestUtils {
 
+    /**
+     * Helper method for parsing specs as a generator would be presented at runtime (inline models resolved, flattened).
+     *
+     * @param specFilePath The path to the specification file
+     * @return A processed OpenAPI document
+     */
+    public static OpenAPI parseFlattenSpec(String specFilePath) {
+        OpenAPI openAPI = new OpenAPIParser().readLocation(specFilePath, null, new ParseOptions()).getOpenAPI();
```

Comment:
Call `TestUtils.parseSpec` instead of reimplementing parsing here?

Reply:
The long term intention is to deprecate parseSpec. For that reason these functions should be independent of each other.

Reply:
Is this request blocking? Why?

Reply:
Sounds good.

Reply:
> The long term intention is to deprecate, possibly remove, or change the location of parseSpec.

Not true. We will still need `parseSpec` for current and future classes which operate on "raw" spec files.

Reply:
Sorry, I meant my long term intention, not ours. @jimschubert classes which operate on raw files aren't something that I know about. What are some examples? Are they not being tested right now?

Reply:
InlineModelResolver, and as mentioned in your other PR, I will be reworking Spec processing using a transformer pipeline which will need this too.

Reply:
Thank you for clarifying that. That other use case is newer and is not yet in the code path.
Per that conversation in the other PR this issue: https://github.com/OpenAPITools/openapi-generator/issues/843 documents the spec processing.

Reply:
parseFlattenSpec has been updated to call parseSpec because we need to keep parseSpec as a public util